.github/workflows/ci-instrumentation-with-services.yml

```diff
@@ -58,30 +46,99 @@ jobs:
           yard: true
           rubocop: true
           build: true
-      - name: "JRuby Filter"
-        id: jruby_skip
-        shell: bash
-        run: |
-          echo "skip=false" >> $GITHUB_OUTPUT
-          [[ "${{ matrix.gem }}" == "bunny"      ]] && echo "skip=true" >> $GITHUB_OUTPUT
-          [[ "${{ matrix.gem }}" == "mysql2"     ]] && echo "skip=true" >> $GITHUB_OUTPUT
-          [[ "${{ matrix.gem }}" == "pg"         ]] && echo "skip=true" >> $GITHUB_OUTPUT
-          [[ "${{ matrix.gem }}" == "que"        ]] && echo "skip=true" >> $GITHUB_OUTPUT
-          [[ "${{ matrix.gem }}" == "racecar"    ]] && echo "skip=true" >> $GITHUB_OUTPUT
-          [[ "${{ matrix.gem }}" == "rdkafka"    ]] && echo "skip=true" >> $GITHUB_OUTPUT
-          [[ "${{ matrix.gem }}" == "redis"      ]] && echo "skip=true" >> $GITHUB_OUTPUT
-          [[ "${{ matrix.gem }}" == "resque"     ]] && echo "skip=true" >> $GITHUB_OUTPUT
-          [[ "${{ matrix.gem }}" == "ruby_kafka" ]] && echo "skip=true" >> $GITHUB_OUTPUT
-          [[ "${{ matrix.gem }}" == "sidekiq"    ]] && echo "skip=true" >> $GITHUB_OUTPUT
-          [[ "${{ matrix.gem }}" == "trilogy"    ]] && echo "skip=true" >> $GITHUB_OUTPUT
-          # This is essentially a bash script getting evaluated, so we need to return true or the whole job fails.
-          true
       - name: "Test JRuby"
-        if: "${{ steps.jruby_skip.outputs.skip == 'false' }}"
         uses: ./.github/actions/test_gem
         with:
           gem:  "opentelemetry-instrumentation-${{ matrix.gem }}"
           ruby: "jruby-9.4.0.0"
+    services:
+      memcached:
+        image: memcached:alpine
+        ports:
+          - 11211:11211
+      mongodb:
+        image: mongo:4.4
+        ports:
+          - 27017:27017
+
+  instrumentation_mysql:
+    strategy:
+      fail-fast: false
+      matrix:
+        gem:
+          - mysql2
+          - trilogy
+        os:
+          - ubuntu-latest
+    name: mysql / ${{ matrix.gem }} / ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: "Test Ruby 3.2"
```

Comment:
Would declaring these steps once with a YAML anchor and then referencing that anchor by alias in the multiple jobs be too DRY?

Reply:
Please add a +1 to this issue: https://github.com/actions/runner/issues/1182

Reply:
Oof.

- Philosophically: not too DRY.
- Realistically: not possible using _only_ GitHub Actions workflow YAML.

OK. 😞

Reply:
Yeah the best we could do was extract the test scripts into a reusable file. That script requires parameters which is what these jobs are supposed to do.